adapter_darwin.go

```diff
@@ -18,7 +19,10 @@ type Adapter struct {
 	peripheralFoundHandler func(*Adapter, ScanResult)
 	scanChan               chan error
 	poweredChan            chan error
-	connectChan            chan cbgo.Peripheral
+
+	// connectChan is a mapping of peripheralId -> chan cbgo.Peripheral,
+	// used to allow multiple callers to call Connect concurrently.
+	connectChan sync.Map
```

Comment:
Since this is no longer a `chan` should perhaps rename the variable to `connectChanMap`.

Reply:
Or perhaps just `connectMap`.